File: app/src/main/java/de/xikolo/views/CustomSizeVideoView.java
package de.xikolo.views;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.VideoView;

public class CustomSizeVideoView extends VideoView {

    private int mForceHeight = 0;
    private int mForceWidth = 0;

    public CustomSizeVideoView(Context context) {
        super(context);
    }

    public CustomSizeVideoView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public CustomSizeVideoView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public void setDimensions(int w, int h) {
        this.mForceHeight = h;
        this.mForceWidth = w;
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        if (mForceHeight > 0 && mForceWidth > 0) {
            setMeasuredDimension(mForceWidth, mForceHeight);
        } else {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        }
    }

}

File: app/src/main/java/de/xikolo/views/CustomSizeVideoView.kt
package de.xikolo.views

import android.content.Context
import android.util.AttributeSet

class CustomSizeVideoView : ExoPlayerVideoView {

    private var forceHeight = 0
    private var forceWidth = 0

    constructor(context: Context) : super(context)

    @JvmOverloads
    constructor(context: Context, attrs: AttributeSet, defStyle: Int = 0) : super(context, attrs, defStyle)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        if (forceHeight > 0 && forceWidth > 0) {
            setMeasuredDimension(forceWidth, forceHeight)
        } else {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        }
    }

}
